src/test/java/codeu/controller/RegisterServletTest.java

```diff
@@ -59,7 +60,7 @@ public void testDoGet() throws IOException, ServletException {
   public void testDoPost_BadUsername() throws IOException, ServletException {
     Mockito.when(mockRequest.getParameter("username")).thenReturn("bad !@#$% username");
```

Comment:
The Mockito setup here is not complete. Check LoginServletTest, you'd need to mock the password, as well as UserStore.